modules/face/samples/sample_train_landmark_detector.cpp

```diff
@@ -0,0 +1,99 @@
+    );
+    //Read in the input arguments
+    if (parser.has("help")){
+        parser.printMessage();
+        cerr << "TIP: Use absolute paths to avoid any problems with the software!" << endl;
+        return 0;
+    }
+    string directory(parser.get<string>("annotations"));
+    //default initialisation
+    Size scale(460,460);
+    scale = Size(parser.get<int>("width"),parser.get<int>("height"));
+    if (directory.empty()){
+        parser.printMessage();
+        cerr << "TIP: Use absolute paths to avoid any problems with the software!" << endl;
+        return -1;
+    }
+    string configfile_name(parser.get<string>("config"));
+    if (configfile_name.empty()){
+        parser.printMessage();
+        cerr << "TIP: Use absolute paths to avoid any problems with the software!" << endl;
+        return -1;
+    }
+    string modelfile_name(parser.get<string>("model"));
+    if (configfile_name.empty()){
+        parser.printMessage();
+        cerr << "TIP: Use absolute paths to avoid any problems with the software!" << endl;
+        return -1;
+    }
+    string cascade_name(parser.get<string>("face_cascade"));
+    if (cascade_name.empty()){
+        parser.printMessage();
+        cerr << "TIP: Use absolute paths to avoid any problems with the software!" << endl;
+        return -1;
+    }
+    //create a vector to store names of files in which annotations
+    //and image names are found
+    /*The format of the file containing annotations should be of following format
+        /data/abc/abc.jpg
+        123.45,345.65
+        321.67,543.89
+        The above format is similar to HELEN dataset which is used for training model
+     */
+    vector<String> filenames;
+    //reading the files from the given directory
+    glob(directory,filenames);
+    //create a vector to store image names
+    vector<String> imagenames;
+    //create object to get landmarks
+    vector< vector<Point2f> > trainlandmarks;
+    //create a pointer to call the base class
+    //pass the face cascade xml file which you want to pass as a detector
+    CascadeClassifier face_cascade;
+    face_cascade.load(cascade_name);
+    Ptr<FacemarkKazemi> facemark= createFacemarkKazemi(face_cascade);
+    //gets landmarks and corresponding image names in both the vectors
+    facemark->getData(filenames,trainlandmarks,imagenames);
+    //vector to store images
+    vector<Mat> trainimages;
+    for(unsigned long i=0;i<imagenames.size();i++){
+        string imgname = imagenames[i].substr(0, imagenames[i].size()-1);
+        string img =string(imgname);
+        Mat src = imread(img);
+        if(src.empty()){
+            cerr<<string("Image"+img+"not found\n.Aborting...")<<endl;
+            return 0;
+        }
+        trainimages.push_back(src);
+    }
+    cout<<"Got data"<<endl;
+    //Now scale data according to the size selected
+    facemark->scaleData(trainlandmarks,trainimages,scale);
+    //calculate mean shape
+    cout<<"Scaled data"<<endl;
```

Comment:
I do not really like a bunch of cout messages in samples. I get the training started one, but the others just clog up the code.